README.md

```diff
@@ -219,4 +219,17 @@ server:
   mapAccounts:
   - "012345678901"
   - "456789012345"
+
+  # map nodes that should conform to the username "system:node:<private-DNS>".  This
+  # requires the authenticator to query the EC2 API in order to discover the private
+  # DNS of the EC2 instance originating the authentication request.  Optionally, you
+  # may specify a role that should be assumed before querying the EC2 API with the
+  # top level key "defaultEC2DescribeInstancesRoleARN"
+  defaultEC2DescribeInstancesRoleARN: arn:aws:iam::000000000000:role/DescribeInstancesRole
+  mapNodes:
```

Comment:
I'm fine with this approach as written, but but what if instead of a special `mapNodes` config, we added a new template variable in `mapRoles` like `{{SessionName}}` but for the instance private DNS name.

Something like:
```yaml
   mapRoles:
   - roleARN: arn:aws:iam::000000000000:role/KubernetesNode
     username: system:node:{{EC2InstancePrivateDNSName}}
     groups:
     - system:nodes
     - system:bootstrappers
```

If that variable was present in the template, we'd do the EC2 DescribeInstances lookup to get the DNS name.

The main improvement would be that it could also be used with other EC2 instance attributes like `{{EC2InstanceVPCID}}`, `{{EC2InstanceSubnetID}}`, `{{EC2InstanceRegion}}`, etc..

Reply:
Gave that approach a go in https://github.com/heptio/authenticator/pull/65